pkg/utils/environment_variable.go

```diff
@@ -0,0 +1,58 @@
+package utils
+
+import (
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+)
+
+var defaultEnvVarValueOrDefault = envVarValueOrDefault
+
+type EnvVarValueOrDefaultFunc func(envVarName string, defaultValue string) string
+
+func IntFromEnvOrDefault(
+	envVarName string,
+	defaultValue string,
+	envVarValueOrDefaultFunc EnvVarValueOrDefaultFunc,
+) (int, error) {
+	if envVarValueOrDefaultFunc == nil {
+		envVarValueOrDefaultFunc = defaultEnvVarValueOrDefault
+	}
+
+	envVarValue := envVarValueOrDefault(envVarName, defaultValue)
+	envVarValueInt, err := strconv.Atoi(envVarValue)
+	if err != nil {
+		return -1, log.RecordedError(log.CAKC010E, envVarName, err.Error())
```

Comment:
return -1 = error or is it return 1?

Reply:
`-1` is the way for me to pass `nil` for an int. it's irrelevant as if `err` is not nil we don't use the returned value.

Reply:
might be worth a comment - it's not a pattern I've seen before. maybe other devs would disagree but @sigalsax already had a question :)